src/modules/frequency/infrastructure/frequency.query.service.prisma.ts

```diff
@@ -39,4 +39,126 @@ export class PrismaFrequencyQueryService implements IFrequencyQueries {
   }
+  async getGeneralAttendance(date: Date): Promise<StudentGeneralAttendanceResponseDTO[]>{
+    const result = await this.prisma.$queryRaw<PrismaStudentGeneralFrequency[]>`
+      WITH 
+      freq_query AS (
+          SELECT
+              id_student AS studentId,
+              'false' AS generalAttendanceAllowed,
+              'PRESENTE' AS status
+          FROM frequency
+          WHERE
+              date = ${date}
+              AND status = 'PRESENTE'
+              AND id_class IS NOT NULL
+          GROUP BY id_student, date
+
+          UNION
+          
+          SELECT
+              id_student AS studentId,
+              'true'    AS generalAttendanceAllowed,
+              'PRESENTE' AS status
+          FROM frequency
+          WHERE
+              date = ${date}
+              AND status = 'PRESENTE'
+              AND id_class IS NULL
+          GROUP BY id_student, date, id
+      ),
+      active_students AS (
+          SELECT
+              id        AS studentId,
+              full_name AS fullName
+          FROM
+              student
+          WHERE
+              status = 'ATIVO'
+      ),
+      left_j AS (
+        SELECT
+            acs.studentId,
+            acs.fullName,
+            fq.generalAttendanceAllowed,
+            fq.status
+        FROM active_students AS acs
+        LEFT JOIN freq_query AS fq
+        ON acs.studentId = fq.studentId
+      )
+      SELECT
+        left_j.studentId,
+        left_j.fullName,
+        ${date} AS date,
+        CASE
+            WHEN left_j.generalAttendanceAllowed IS NULL THEN 'true'
+            ELSE left_j.generalAttendanceAllowed
+        END AS generalAttendanceAllowed,
+        CASE
+              WHEN left_j.status IS NULL THEN 'AUSENTE'
+              ELSE left_j.status
+        END AS status
+      FROM left_j       
+    `;
+    if(!result){
+      return []
+    };
+    const attendanceArray = result.map(arg=>FrequencyDTOMapper.toStudentGeneralAttendanceDTO(arg));
+    return attendanceArray;
+  }
+  async getStudentByClassAndDateAttendanceList(classId: number, date: Date) {
+    const results = await this.prisma.$queryRaw<PrismaStudentClassAttendance[]>`
+        WITH
+        total_classes AS (
+          SELECT COUNT(DISTINCT date) AS count
+          FROM frequency
+          WHERE id_class = ${classId}
+        ),
+        student_attendance AS (
+          SELECT
+            id_student,
+            COUNT(*) AS present_count
+          FROM frequency
+          WHERE id_class = ${classId}
+          AND status = 'PRESENTE'
+          GROUP BY id_student
+        ),
+        daily_attendance AS (
+          SELECT
+            id,
+            id_student,
+            status,
+            notes
+          FROM frequency
+          WHERE id_class = ${classId}
+          AND date = ${date}
+        )
+        SELECT
+          d.id AS "frequencyId",
+          s.id AS "stutentId",
```

Comment:
typo. Não sei se algo quebrou depois da última sugestão e o campo tem que estar assim.
```suggestion
          s.id AS "studentId",
```

Reply:
Alterarei isso diretamente no código.